c_sources/osqp_mex.cpp

```diff
@@ -253,9 +253,7 @@ void mexFunction(int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[])
 
     // delete the object and its data
     if (!strcmp("delete", cmd)) {
```

Comment:
I'm a bit confused about this. During the constructor ("new"), I dynamically create a new OSQPCodegenDefines object (see Line 159). Seeing how osqp_cleanup destroys osqpData->solver, I thought I would have to free osqpData->defines manually. However, when I did that, the program crashed because destroyObject tries to destroy the already freed memory. What I don't understand is why this doesn't happen with osqpData->solver, which is freed by osqp_cleanup.